docs/libauthor_docs.rst

```diff
@@ -0,0 +1,65 @@
+.. currentmodule:: uarray
+
+.. _libauthordocs:
+
+Documentation for backend providers
+===================================
+
+Backend providers can provide a back-end for a defined API within
+the :obj:`uarray` ecosystem. To find out how to define your own
+API with :obj:`uarray`, see :ref:`mmauthordocs`. To find out how
+your backend will be provided, use :ref:`userdocs`.
+
+Backend providers need to be aware of three protocols: ``__ua_domain__``,
+``__ua_function__`` and ``__ua_convert__``. These three must be a part of
+every backend object.
+
+``__ua_domain__``
+-----------------
+
+``__ua_domain__`` is a string containing the domain of the backend. This is,
+by convention, the name of the module (or one of its dependencies or parents)
+that contain the multimethods. For example, ``scipy`` and ``numpy.fft`` could
+both use the ``numpy`` domain.
+
+``__ua_function__``
+-------------------
+
+This is the most important protocol, one that defines the implementation of a
+multimethod. It has the signature ``(method, args, kwargs, dispatchable_args)``.
```

Comment:
It's not clear to me what a backend is meant to do with `dispatchable_args`. Isn't the job of `argument_replacer` to rewrite the function call arguments such that this doesn't matter?

Reply:
When `__ua_convert__` is not present, the purpose is to recognize whether these arguments can be handled or not.

Reply:
Is it not enough for them to be in `args` and `kwargs`?

Reply:
You don't know which ones in `args`/`kwargs` are dispatchable and which ones are standard arguments, for a generalized function call. I have added documentation to that effect. I also made `__ua_convert__` mandatory.

Reply:
If you prefer this functionality be moved into `__ua_convert__`, that'd be fine by me too.

Reply:
I think it's better to have a clear seperation of responsibilities here.

Reply:
Done!